### ISSUE TITLE
Document postcss-flexbugs-fixes in declaration-block-no-redundant-longhand-properties

### ISSUE THREAD

**Issue by AlexWayfer** [stylelint/stylelint#3144]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

A dangerous behavior (a bug, I guess, or false positive): `declaration-block-no-redundant-longhand-properties` triggers for `flex`, where `flex-basis` is `calc()`, [which doesn't work correctly in IE 10–11](https://github.com/philipwalton/flexbugs#flexbug-8).

> Which rule, if any, is this issue related to?

`declaration-block-no-redundant-longhand-properties`

> What CSS is needed to reproduce this issue?

e.g.

```css
.foo {
  flex-basis: calc(100% / 3);
  flex-grow: 1;
  flex-shrink: 0;
}
```

> What stylelint configuration is needed to reproduce this issue?

e.g.

```json
{
  "rules": {
    "declaration-block-no-redundant-longhand-properties": true
  }
}
```

> Which version of stylelint are you using?

`8.4.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

CLI with `yarn run stylelint --config .stylelintrc path/to/file.scss`"

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

No. It's reproducable with regular CSS.

> What did you expect to happen?

No warnings to be flagged.

> What actually happened (e.g. what warnings or errors you are getting)?

The following warnings were flagged:

```shell
 4:2  ✖  Expected shorthand property "flex"                                                                      declaration-block-no-redundant-longhand-properties
```

**Comment by evilebottnawi** [comment]
@AlexWayfer Thanks for issue, i think we can add option and/or look in browserslist config to avoid this problem. If you don't want support IE10-11 rule works as expected.

**Comment by AlexWayfer** [comment]
> i think we can add option and/or look in browserslist config to avoid this problem

Yes, I'm adding `IE 11` into `browserslist` with `stylelint-no-unsupported-browser-features` plugin and this issue is still here.

**Comment by CAYdenberg** [comment]
I believe the rule is behaving as expected.

According to the specification, that CSS need not be written longhand, and the rule is correctly stating that. In effect, writing out these declarations is a hack to avoid IE's shoddy implementation of flexbox.

I think this could be avoided with e.g. `ignoreShorthands: ["/flex-/"]` - possibly we could expand documentation to suggest that?

**Comment by AlexWayfer** [comment]
> According to the specification, that CSS need not be written longhand

But may be.

> writing out these declarations is a hack to avoid IE's shoddy implementation of flexbox

Or work-around. "Hack" is something unregular, offending.

> ignoreShorthands: ["/flex-/"] - possibly we could expand documentation to suggest that?

Maybe. But can the `stylelint-no-unsupported-browser-features` plugin do something similar automatically with `IE 11` in `browserslist`?

**Comment by jeddy3** [comment]
I agree with @CAYdenberg.

The philosophy of stylelint is to help authors write the nicest code possible because, in this day and age, there is a lot of tooling to help transform this nice code into not so nice code that works around browser issues. 

This is why we have all the `*-no-vendor-prefix` rules... as stylelint encourages the use of autoprefixer.

In this instance, I recommend using [postcss-flexbugs-fixes](https://github.com/luisrudge/postcss-flexbugs-fixes) to [transform your nicely authored code](https://github.com/luisrudge/postcss-flexbugs-fixes#bug-81a) into something that is compatible with IE11.

This is very much the happy path here. However, you can, as @CAYdenberg suggested, make use of the `ignoreShorthands` secondary option if you want to managed this IE bug by hand.

I suggest we add a link to postcss-flexbugs-fixes in the `declaration-block-no-redundant-longhand-properties` rule documentation to help other users that run into this issue.
